### PR TITLE
PR: Add a debug current line or selection button

### LINKED ISSUE #17888: Add Debug line functionality to Debug menu and toolbar
So Spyder's menu bar has Run and Debug menus which have items for running/debugging a file and running/debugging a cell, and running the current line, but not debugging the current line. I propose adding this menu item to the Debug menu bar. Similar to running a single line, which pastes the current line into the IPython console and executes, debugging a single line can paste the line into the IPython console prepended with "debug" and thus invoke the debugger for executing the single line of code.

### PR DESCRIPTION
**Pull request by impact27**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #17888 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

File: spyder/plugins/editor/widgets/editor.py [hunk 1]
```diff
@@ -2819,15 +2819,19 @@ def run_selection(self):
...
         """
+        if prefix is None:
+            prefix = ''
         text = self.get_current_editor().get_selection_as_executable_code()
         if text:
-            self.exec_in_extconsole.emit(text.rstrip(), self.focus_to_editor)
+            self.exec_in_extconsole.emit(
+                prefix + text.rstrip(), self.focus_to_editor)
             return
         editor = self.get_current_editor()
```
Comment (ccordoba12) [inline spyder/plugins/editor/widgets/editor.py:2829]:
```suggestion

        editor = self.get_current_editor()
```

**Review by ccordoba12 — COMMENTED** [review]
Thanks @impact27 for this! Apart from the very minor suggestions I left for you, I think the only thing missing here is a test that checks this functionality is working as expected and that it won't break in the future.

Please add that test to `test_mainwindow.py` to verify that clicking on the new "Debug selection" starts a debugging session with the right code.

**Comment by impact27** [comment]
@ccordoba12 the failures appear to be unrelated to this PR

**Review by ccordoba12 — APPROVED** [review]
Looks good to me now, thanks @impact27!

**Pull request merged**